src/patterns.rst

```diff
@@ -1350,6 +1362,25 @@ proceeds as follows:
 Only the :t:`[binding]s` of a matched :t:`pattern-without-alternation` are
 introduced into a :t:`binding scope`.
 
+
+.. _fls_VsBXBj4AqCj1:
+
+Or-pattern Matching
+~~~~~~~~~~~~~~~~~~~
+
+
+:dp:`fls_njpiXkgi8Ryb`
+Or-pattern matching of an :t:`or-pattern` of the form `constructor(or-pattern, rest)`,
+where `constructor` is an arbitrary constructor, `or-pattern` is the :t:`or-pattern`,
+and `rest` is optionally a remaining pattern, proceeds as follows:
+
+#. :dp:`fls_nE7qZpHy9TPu`
+   Perform pattern matching of the form `constructor(subpattern, rest)`, where
+   `subpattern` is a :t:`subpattern` of the :t:`or-pattern`, starting from the
+   first such :t:`subpattern` and proceeding in declarative order.
+#. :dp:`fls_P8yB2b5enpw7`
+   Otherwise pattern matching fails.
```

Comment:
```suggestion
#. :dp:`fls_nE7qZpHy9TPu`
   Perform pattern matching of the form `constructor(subpattern, rest)`, where
   `subpattern` is a :t:`subpattern` of the :t:`or-pattern`, starting from the
   first such :t:`subpattern` and proceeding in declarative order.

#. :dp:`fls_P8yB2b5enpw7`
   Otherwise pattern matching fails.
```
(reads a little better at the source level)